### ISSUE TITLE
Add searching websites like google, youtube, like KISS and Tinybit launcher.

### ISSUE THREAD

**Issue by magnus-ISU** [kennethnym/StarlightLauncher#26]
Also support adding custom URLs.

Tinybit also allows providing icons when adding a custom launcher. I think that would be nice and then you could show search options in a horizontal list like apps are now, which would be nice.

**Comment by CyberaxAnnon** [comment]
DuckDuckgo - https://duckduckgo.com/
Brave search - https://search.brave.com/
Startpage - https://www.startpage.com/
SearchX - https://searx.thegpm.org/

These would also be good to implement for privacy conscious people if possible :)

**Comment by kennethnym** [comment]
Yes, this is planned - you will be able to select the search engine of your choice.

**Comment by daffydock** [comment]
Perhaps, if privacy is a focus then Starpage should *not* be added, as it was bought by System1. A well known target marketing company.   

https://restoreprivacy.com/startpage-system1-privacy-one-group/

**Comment by ayecptn** [comment]
I think it should be up to the user if he wants a privacy-focused web search. I would let user choose from ~5 different popular  search engines and a field where the user can add a custom engine.

@kennethnym I have already implemented a basic web search module, which just uses duckduckgo hardcoded. You can check it out on my fork in the WebSearch branch. Here's a small demo:
[starlight_websearch.webm](https://user-images.githubusercontent.com/61555365/188709052-14a267c3-a197-4d85-b054-ef10e9fa377e.webm)

**Comment by kennethnym** [comment]
That's awesome! I am actually researching how to fetch search suggestions for a given query, seems like there's no direct way of doing so.

**Comment by kennethnym** [comment]
Fun fact: in a really early prototype of Starlight (It used to be called Spotlight!) I was actually able to use DDG's api to fetch definitions for a given search query, for example if you type in "elephant" the launcher will fetch the definition from DDG and show it in the launcher, but I scraped it because personally I don't use it that much. But seeing there's a request for it, I might be able to bring it back.

**Comment by ayecptn** [comment]
Oh, so you mean fetching and displaying the search results directly in the launcher? Would be nice, but it could be a problem as not every search engine offers an public api.

**Comment by kennethnym** [comment]
@ayecptn Yes, afaik only DuckDuckGo has a public API for knowledge search. It can be a separate search module though. For the web search module your implementation is what I have in mind. I am open for a PR - I'm planning to scrape the monetization plan and make this a community project (donations still welcome!)

**Comment by ayecptn** [comment]
Sounds like a good  idea to make two different modules.
I would have done a PR after I implemented the setting to choose from different search engines, but I am to lazy/unmotivated to do anything coding related currently.
But I can also do a PR with the current hardcoded implementation and then a second PR with the settings if you want.

**Comment by kennethnym** [comment]
Don't worry about it - if it is just a simple button for triggering web search then it is easy to implement.
